From c1557c5c0e2b12a5a6300e0c51592f5606b386ce Mon Sep 17 00:00:00 2001
From: Bryan O'Donoghue <bryan.odonoghue@linaro.org>
Date: Tue, 6 Aug 2019 12:38:32 +0100
Subject: [PATCH 04/13] mx7_common: Disjunct on CONFIG_OPTEE_SKIP_LOWLEVEL_INIT

Switch from complex ifdef chain to more simple disjunction on
CONFIG_OPTEE_SKIP_LOWLEVEL_INIT to determine if SKIP_LOWLEVEL_INIT should
be on.

Signed-off-by: Bryan O'Donoghue <bryan.odonoghue@linaro.org>
---
 configs/pico-imx7d_bl33_defconfig | 5 ++++-
 configs/warp7_bl33_defconfig      | 1 +
 include/configs/mx7_common.h      | 9 +++------
 3 files changed, 8 insertions(+), 7 deletions(-)

diff --git a/configs/pico-imx7d_bl33_defconfig b/configs/pico-imx7d_bl33_defconfig
index 7c559cfec4..5ebecd290f 100644
--- a/configs/pico-imx7d_bl33_defconfig
+++ b/configs/pico-imx7d_bl33_defconfig
@@ -64,4 +64,7 @@ CONFIG_USB_ETH_CDC=y
 CONFIG_USBNET_HOST_ADDR="de:ad:be:af:00:00"
 CONFIG_VIDEO=y
 CONFIG_OF_LIBFDT_OVERLAY=y
-CONFIG_OPTEE_TZDRAM_SIZE=0x2000000
+CONFIG_OPTEE=y
+CONFIG_OPTEE_TZDRAM_BASE=0x9e000000
+CONFIG_OPTEE_TZDRAM_SIZE=0x02000000
+CONFIG_SKIP_LOWLEVEL_INIT=y
diff --git a/configs/warp7_bl33_defconfig b/configs/warp7_bl33_defconfig
index 0f28d5881d..f20aa1c4bc 100644
--- a/configs/warp7_bl33_defconfig
+++ b/configs/warp7_bl33_defconfig
@@ -57,6 +57,7 @@ CONFIG_USBNET_HOST_ADDR="de:ad:be:af:00:00"
 CONFIG_OPTEE=y
 CONFIG_OPTEE_TZDRAM_BASE=0x9e000000
 CONFIG_OPTEE_TZDRAM_SIZE=0x02000000
+CONFIG_SKIP_LOWLEVEL_INIT=y
 CONFIG_FIT=y
 CONFIG_FIT_SIGNATURE=y
 CONFIG_FIT_VERBOSE=y
diff --git a/include/configs/mx7_common.h b/include/configs/mx7_common.h
index 4f822ef9a0..44beefe816 100644
--- a/include/configs/mx7_common.h
+++ b/include/configs/mx7_common.h
@@ -55,14 +55,11 @@
 #endif
 
 /*
- * If we have defined the OPTEE ram size and not OPTEE it means that we were
- * launched by OPTEE, because of that we shall skip all the low level
- * initialization since it was already done by ATF or OPTEE
+ * If OPTEE_SKIP_LOWLEVEL_INIT is flagged indicate CONFIG_SKIP_LOWLEVEL_INIT and
+ * subsequently skip over reserved a range of reserved low-level bit twiddling.
  */
-#if (CONFIG_OPTEE_TZDRAM_SIZE != 0)
-#ifndef CONFIG_OPTEE
+#ifdef OPTEE_SKIP_LOWLEVEL_INIT
 #define CONFIG_SKIP_LOWLEVEL_INIT
 #endif
-#endif
 
 #endif
-- 
2.22.0

